### PR TITLE
Refactor build file to sections

### PR DESCRIPTION
**Pull request by arturbosch**

### CODE REVIEW

**Comment by arturbosch** [comment]
Codecov gets triggered before any report can be published?

**Comment by schalkms** [comment]
@arturbosch maybe because of the old setting removed in #2546 ?

File: build.gradle.kts [hunk 1]
```diff
@@ -267,6 +267,16 @@ subprojects {
...
 
+val analysisDir = files(projectDir)
+val baselineFile = file("$rootDir/config/detekt/baseline.xml")
+val formatConfigFile = files("$rootDir/config/detekt/format.yml")
+val configFiles = files("$rootDir/config/detekt/detekt.yml")
+
+val kotlinFiles = "**/*.kt"
+val kotlinScriptFiles = "**/*.kts"
+val resourceFiles = "**/resources/**"
+val buildFiles = "**/build/**"
```
Comment (BraisGabin) [inline build.gradle.kts]: Can these be `const`?

Comment (arturbosch) [reply build.gradle.kts]: Unfortunately not as they get compiled to locale properties inside a script object.

**Comment by schalkms** [comment]
Coverage went down to 25% 😐

**Comment by arturbosch** [comment]
And some of the newer PRs increase it by x.xx% without writing tests :S

**Pull request merged**